nncf/experimental/tensor/functions.py

```diff
@@ -324,28 +322,105 @@ def zeros_like(a: TTensor) -> TTensor:
     return Tensor(zeros_like(a.data))
 
 
-__all__ = [
-    "device",
-    "squeeze",
-    "flatten",
-    "max",
-    "min",
-    "abs",
-    "astype",
-    "reshape",
-    "all",
-    "allclose",
-    "any",
-    "count_nonzero",
-    "isempty",
-    "isclose",
-    "maximum",
-    "minimum",
-    "ones_like",
-    "minimum",
-    "where",
-    "zeros_like",
-]
+@functools.singledispatch
+def stack(x: List[TTensor], axis: int = 0) -> TTensor:
+    """
+    Stacks a list or deque of Tensors rank-R tensors into one Tensor rank-(R+1) tensor.
+
+    :param x: List or deque of Tensors.
+    :param axis: The axis to stack along.
+    :return: Stacked Tensor.
+    """
+    if isinstance(x, List):
+        unwrapped_x = [i.data for i in x]
+        # singledispatch cannot dispatch function by element in a list
+        res = stack.dispatch(type(unwrapped_x[0]))(unwrapped_x, axis=axis)
+        return Tensor(res)
+    raise NotImplementedError(f"Function `stack` is not implemented for {type(x)}")
+
+
+@functools.singledispatch
+@_tensor_guard
+def unstack(a: TTensor, axis: int = 0) -> List[Tensor]:
```

Comment:
I guess that here, and everywhere, the type hint for the function args should be `Tensor`, not `TTensor`

Reply:
> I guess that here, and everywhere, the type hint for the function args should be `Tensor`, not `TTensor`

I think this is not entirely correct. I mean these functions return a `Tensor` object only if there is no registered version for the type of the passed argument.

Reply:
Updated all typhints `TTensor` now only used for .data